hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/RowIndexEncoderV1.java

```diff
@@ -45,6 +46,8 @@ public int write(Cell cell) throws IOException {
         startOffset = out.size();
       }
       rowsOffsetBAOS.writeInt(out.size() - startOffset);
+      // added for the int written in the previous line
+      context.getEncodingState().postCellEncode(0, Bytes.SIZEOF_INT);
```

Comment:
Hmm.. this looks bit ugly...
We can do postCellEncode() only here
Anyway the above is encode call within codec and the actual encode happens here. Once that is done,  u can call postCellEncode() with encodedsize been calculated